code/game/gamemodes/clock_cult/clock_helpers/component_helpers.dm

```diff
@@ -62,6 +62,10 @@
 		else
 			return null
 
+//returns a component icon from a component id
+/proc/get_component_icon(id)
+	return "<img style='width:16px; height:16px' src='data:image/png;base64,[icon2base64(icon('icons/obj/tgui_components.dmi', id))]'/>"
```

Comment:
I feel like you should be cacheing this for some reason but I dunno, @RemieRichards?